### ISSUE TITLE
[BUG] [2.9] Unable to provision/import hosted provider clusters 

### ISSUE THREAD

**Issue by valaparthvi** [rancher/rancher#46152]
**Rancher Server Setup**
- Rancher version: dev (?) Used 2.9-head
- Installation option (Docker install/Helm Chart): 
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): Helm chart on K3s
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: 1.28.9+k3s1
- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Provisioned and imported hosted provider

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: Admin

**Describe the bug**

Unable to provision or import a hosted provider successfully due to the following error:
```
Error while applying agent YAML, it will be retried automatically: exit status 1, clusterrole.rbac.authorization.k8s.io/proxy-clusterrole-kubeapiserver unchanged clusterrolebinding.rbac.authorization.k8s.io/proxy-role-binding-kubernetes-master unchanged namespace/cattle-system unchanged serviceaccount/cattle unchanged clusterrolebinding.rbac.authorization.k8s.io/cattle-admin-binding unchanged secret/cattle-credentials-8e1d65d unchanged clusterrole.rbac.authorization.k8s.io/cattle-admin unchanged error: error parsing ./management-statefile_path_redacted: error converting YAML to JSON: yaml: line 105: mapping values are not allowed in this context
```
**To Reproduce**

1. Provision or import an EKS/GKE/AKS cluster and wait for it to become Active.

**Result**
The cluster is not provisioned/imported and returns an error.

**Expected Result**

The cluster should be provisioned/imported successfully.

**Screenshots**

![Screenshot from 2024-07-15 11-01-02](https://github.com/user-attachments/assets/56d64352-8a15-455d-90df-cce6cfde4e22)

**Additional context**

**Comment by cpinjani** [comment]
Cluster provisioning logs:
```
2024/07/15 06:52:57 [ERROR] error syncing 'c-xbh8p': handler gke-operator-controller: waiting gke crd to be initialized, cluster: c-xbh8p, requeuing
2024/07/15 06:52:57 [ERROR] error syncing 'c-xbh8p': handler gke-operator-controller: waiting gke crd to be initialized, cluster: c-xbh8p, requeuing
2024/07/15 06:52:57 [ERROR] error syncing 'c-xbh8p': handler gke-operator-controller: waiting gke crd to be initialized, cluster: c-xbh8p, requeuing
2024/07/15 06:53:02 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:02 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:02 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:02 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:02 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:02 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:02 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:02 [INFO] waiting for cluster create [c-xbh8p] to start
W0715 06:53:02.598495      39 warnings.go:70] v1 ComponentStatus is deprecated in v1.19+
2024/07/15 06:53:02 [INFO] Watching metadata for gke.cattle.io/v1, Kind=GKEClusterConfig
2024/07/15 06:53:04 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:04 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:07 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:07 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:12 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:12 [INFO] waiting for cluster create [c-xbh8p] to start
2024/07/15 06:53:17 [INFO] setting initial upstreamSpec on cluster [c-xbh8p]
2024/07/15 06:53:17 [INFO] waiting for cluster GKE [c-xbh8p] to finish creating
2024/07/15 06:53:17 [INFO] waiting for cluster GKE [c-xbh8p] to finish creating
2024/07/15 06:53:17 [INFO] waiting for cluster GKE [c-xbh8p] to finish creating
2024/07/15 06:53:22 [INFO] waiting for cluster GKE [c-xbh8p] to finish creating
.
.
.
2024/07/15 06:58:32 [INFO] waiting for cluster GKE [c-xbh8p] to finish creating
2024/07/15 06:58:37 [INFO] waiting for cluster GKE [c-xbh8p] to finish creating
2024/07/15 06:58:42 [INFO] waiting for cluster GKE [c-xbh8p] to finish creating
2024/07/15 06:58:47 [ERROR] error syncing 'c-xbh8p': handler cluster-deploy: secrets "cattle-global-data/" not found, requeuing
2024/07/15 06:58:47 [ERROR] error syncing '_all_': handler user-controllers-controller: userControllersController: failed to set peers for key _all_: failed to start user controllers for cluster c-xbh8p: secrets "cattle-global-data/" not found, requeuing
2024/07/15 06:58:47 [ERROR] error syncing '_all_': handler user-controllers-controller: userControllersController: failed to set peers for key _all_: failed to start user controllers for cluster c-xbh8p: secrets "cattle-global-data/" not found, requeuing
2024/07/15 06:58:47 [ERROR] error syncing '_all_': handler user-controllers-controller: userControllersController: failed to set peers for key _all_: failed to start user controllers for cluster c-xbh8p: secrets "cattle-global-data/" not found, requeuing
2024/07/15 06:58:47 [ERROR] error syncing '_all_': handler user-controllers-controller: userControllersController: failed to set peers for key _all_: failed to start user controllers for cluster c-xbh8p: secrets "cattle-global-data/" not found, requeuing
2024/07/15 06:58:47 [ERROR] error syncing '_all_': handler user-controllers-controller: userControllersController: failed to set peers for key _all_: failed to start user controllers for cluster c-xbh8p: secrets "cattle-global-data/" not found, requeuing
2024/07/15 06:58:47 [INFO] EnsureSecretForServiceAccount: waiting for secret [kontainer-engine-token-rzq5l] to be populated with token
2024/07/15 06:58:47 [ERROR] error syncing 'c-xbh8p': handler cluster-deploy: secrets "cattle-global-data/" not found, handler gke-operator-controller: resource name may not be empty, requeuing
2024/07/15 06:58:47 [ERROR] error syncing 'c-xbh8p': handler cluster-deploy: secrets "cattle-global-data/" not found, requeuing
2024/07/15 06:58:47 [INFO] Starting cluster controllers for c-xbh8p
2024/07/15 06:58:47 [INFO] Starting /v1, Kind=ResourceQuota controller
2024/07/15 06:58:47 [INFO] Starting /v1, Kind=Namespace controller
2024/07/15 06:58:47 [INFO] Starting /v1, Kind=LimitRange controller
2024/07/15 06:58:47 [INFO] Starting rbac.authorization.k8s.io/v1, Kind=Role controller
2024/07/15 06:58:47 [INFO] Starting rbac.authorization.k8s.io/v1, Kind=RoleBinding controller
2024/07/15 06:58:47 [INFO] Starting /v1, Kind=Node controller
2024/07/15 06:58:47 [INFO] Starting /v1, Kind=ServiceAccount controller
2024/07/15 06:58:47 [INFO] Starting /v1, Kind=Secret controller
2024/07/15 06:58:47 [INFO] Starting apiregistration.k8s.io/v1, Kind=APIService controller
2024/07/15 06:58:47 [INFO] Starting rbac.authorization.k8s.io/v1, Kind=ClusterRole controller
2024/07/15 06:58:47 [INFO] Starting rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding controller
2024/07/15 06:58:47 [INFO] Starting cluster agent for c-xbh8p [owner=true]
2024/07/15 06:58:47 [INFO] Creating clusterRole for roleTemplate Project Owner (project-owner).
2024/07/15 06:58:47 [INFO] Creating clusterRole for roleTemplate Cluster Owner (cluster-owner).
2024/07/15 06:58:47 [INFO] Creating clusterRole for roleTemplate Cluster Owner (cluster-owner).
2024/07/15 06:58:47 [INFO] Created machine for node [gke-cpinjani-gke-group-1-dd8ef298-jlxl]
2024/07/15 06:58:47 [INFO] Creating clusterRole for roleTemplate Project Owner (project-owner).
2024/07/15 06:58:47 [INFO] Creating clusterRole for roleTemplate Project Owner (project-owner).
2024/07/15 06:58:47 [INFO] Creating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:47 [INFO] Creating user for principal system://c-xbh8p
2024/07/15 06:58:47 [INFO] Creating globalRoleBindings for u-bdggpngcxq
2024/07/15 06:58:47 [INFO] Creating clusterRoleBinding User user-gckwh Role cluster-owner
2024/07/15 06:58:47 [INFO] Creating system token for u-bdggpngcxq, token: agent-u-bdggpngcxq
2024/07/15 06:58:48 [INFO] Creating new GlobalRoleBinding for GlobalRoleBinding grb-snfpx
2024/07/15 06:58:48 [INFO] [mgmt-auth-grb-controller] Creating clusterRoleBinding for globalRoleBinding grb-snfpx for user u-bdggpngcxq with role cattle-globalrole-user
2024/07/15 06:58:48 [INFO] [mgmt-auth-crtb-controller] Creating clusterRoleBinding for membership in cluster c-xbh8p for subject u-bdggpngcxq
2024/07/15 06:58:48 [INFO] [mgmt-auth-crtb-controller] Creating roleBinding for subject u-bdggpngcxq with role cluster-owner in namespace c-xbh8p
2024/07/15 06:58:48 [INFO] [mgmt-auth-crtb-controller] Creating roleBinding for subject u-bdggpngcxq with role cluster-owner in namespace p-m2pjb
2024/07/15 06:58:48 [INFO] [mgmt-auth-crtb-controller] Creating roleBinding for subject u-bdggpngcxq with role cluster-owner in namespace p-ws6lt
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Creating clusterRoleBinding User u-bdggpngcxq Role cluster-owner
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] EnsureSecretForServiceAccount: waiting for secret [cattle-impersonation-user-gckwh-token-c47vk] to be populated with token
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role project-owner in default
2024/07/15 06:58:48 [INFO] EnsureSecretForServiceAccount: waiting for secret [cattle-impersonation-user-gckwh-token-d8p9m] to be populated with token
2024/07/15 06:58:48 [INFO] EnsureSecretForServiceAccount: waiting for secret [cattle-impersonation-u-bdggpngcxq-token-lwwmb] to be populated with token
2024/07/15 06:58:48 [INFO] EnsureSecretForServiceAccount: waiting for secret [cattle-impersonation-u-bdggpngcxq-token-t6gkz] to be populated with token
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [ERROR] defaultSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=kube-node-lease, err=Operation cannot be fulfilled on namespaces "kube-node-lease": the object has been modified; please apply your changes to the latest version and try again
2024/07/15 06:58:48 [INFO] Creating clusterRole for roleTemplate Create Namespaces (create-ns).
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Creating clusterRole for roleTemplate Create Namespaces (create-ns).
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role admin in default
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [ERROR] defaultSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=kube-public, err=Operation cannot be fulfilled on namespaces "kube-public": the object has been modified; please apply your changes to the latest version and try again
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role admin in kube-node-lease
2024/07/15 06:58:48 [ERROR] defaultSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=cattle-impersonation-system, err=Operation cannot be fulfilled on namespaces "cattle-impersonation-system": the object has been modified; please apply your changes to the latest version and try again
2024/07/15 06:58:48 [ERROR] defaultSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=cattle-system, err=Operation cannot be fulfilled on namespaces "cattle-system": the object has been modified; please apply your changes to the latest version and try again
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role project-owner in kube-public
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role project-owner in kube-node-lease
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role admin in kube-public
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role admin in kube-public
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role project-owner in kube-system
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role project-owner in cattle-system
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role admin in cattle-system
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Updating clusterRole project-owner-promoted for project access to global resource.
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role admin in kube-system
2024/07/15 06:58:48 [INFO] Creating clusterRoleBinding for project access to global resource for subject user-gckwh role create-ns.
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role admin in cattle-system
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role admin in cattle-impersonation-system
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role project-owner in cattle-impersonation-system
2024/07/15 06:58:48 [INFO] Creating clusterRoleBinding for project access to global resource for subject user-gckwh role create-ns.
2024/07/15 06:58:48 [INFO] Creating roleBinding User user-gckwh Role admin in cattle-impersonation-system
2024/07/15 06:58:48 [INFO] Creating clusterRoleBinding for project access to global resource for subject user-gckwh role p-m2pjb-namespaces-edit.
2024/07/15 06:58:48 [INFO] Updating clusterRoleBinding crb-twrl2idna3 for project access to global resource for subject user-gckwh role create-ns.
2024/07/15 06:58:48 [INFO] Creating clusterRoleBinding for project access to global resource for subject user-gckwh role p-ws6lt-namespaces-edit.
2024/07/15 06:58:48 [INFO] Creating clusterRoleBinding for project access to global resource for subject user-gckwh role p-m2pjb-namespaces-edit.
2024/07/15 06:58:49 [INFO] Creating clusterRoleBinding for project access to global resource for subject user-gckwh role project-owner-promoted.
2024/07/15 06:58:49 [INFO] Creating clusterRoleBinding for project access to global resource for subject user-gckwh role project-owner-promoted.
2024/07/15 06:58:49 [ERROR] defaultSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=cattle-impersonation-system, err=Operation cannot be fulfilled on namespaces "cattle-impersonation-system": the object has been modified; please apply your changes to the latest version and try again
2024/07/15 06:58:49 [INFO] Updating clusterRoleBinding crb-r5i6aj2vfz for project access to global resource for subject user-gckwh role project-owner-promoted.
2024/07/15 06:58:49 [INFO] Updating clusterRoleBinding crb-r5i6aj2vfz for project access to global resource for subject user-gckwh role project-owner-promoted.
2024/07/15 06:58:49 [ERROR] defaultSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=cattle-impersonation-system, err=Operation cannot be fulfilled on namespaces "cattle-impersonation-system": the object has been modified; please apply your changes to the latest version and try again
2024/07/15 06:59:16 [INFO] Creating system token for u-bdggpngcxq, token: agent-u-bdggpngcxq
2024/07/15 06:59:44 [INFO] Creating system token for u-bdggpngcxq, token: agent-u-bdggpngcxq
2024/07/15 07:00:12 [INFO] Creating system token for u-bdggpngcxq, token: agent-u-bdggpngcxq
2024/07/15 07:00:40 [INFO] Creating system token for u-bdggpngcxq, token: agent-u-bdggpngcxq
2024/07/15 07:01:08 [ERROR] error syncing 'c-xbh8p': handler cluster-deploy: Error while applying agent YAML, it will be retried automatically: exit status 1, clusterrole.rbac.authorization.k8s.io/proxy-clusterrole-kubeapiserver unchanged clusterrolebinding.rbac.authorization.k8s.io/proxy-role-binding-kubernetes-master unchanged namespace/cattle-system unchanged serviceaccount/cattle unchanged clusterrolebinding.rbac.authorization.k8s.io/cattle-admin-binding unchanged secret/cattle-credentials-e237644 unchanged clusterrole.rbac.authorization.k8s.io/cattle-admin unchanged error: error parsing ./management-state/tmp/yaml-3748234888: error converting YAML to JSON: yaml: line 105: mapping values are not allowed in this context , requeuing
2024/07/15 07:01:08 [INFO] Creating system token for u-bdggpngcxq, token: agent-u-bdggpngcxq
2024/07/15 07:01:36 [INFO] Creating system token for u-bdggpngcxq, token: agent-u-bdggpngcxq
2024/07/15 07:02:04 [ERROR] error syncing 'c-xbh8p': handler cluster-deploy: Error while applying agent YAML, it will be retried automatically: exit status 1, clusterrole.rbac.authorization.k8s.io/proxy-clusterrole-kubeapiserver unchanged clusterrolebinding.rbac.authorization.k8s.io/proxy-role-binding-kubernetes-master unchanged namespace/cattle-system unchanged serviceaccount/cattle unchanged clusterrolebinding.rbac.authorization.k8s.io/cattle-admin-binding unchanged secret/cattle-credentials-e237644 unchanged clusterrole.rbac.authorization.k8s.io/cattle-admin unchanged error: error parsing ./management-state/tmp/yaml-3086159190: error converting YAML to JSON: yaml: line 105: mapping values are not allowed in this context , requeuing
```

**Comment by thehejik** [comment]
Fleet v2.9-head e2e testing also affected when importing clusters into rancher.

Seems there is now a need to install Rancher `v2.9-head` with these additional helm flags:
```
--set extraEnv[0].name=CATTLE_AGENT_IMAGE \
--set extraEnv[0].value=rancher/rancher-agent:v2.9-head
```
otherwise the image tag is not present in rancher import yaml for the `cattle-cluster-agent` Deployment resource and downstream cluster import will fail.
```yaml
...
          - name: CATTLE_INGRESS_IP_DOMAIN
            value: sslip.io
          - name: STRICT_VERIFY
            value: "true"
          image: rancher/rancher-agent:    <---- HERE
          volumeMounts:
          - name: cattle-credentials
            mountPath: /cattle-credentials
            readOnly: true
...
```
Not sure this is expected but few days back the extra flags were not needed.

**Comment by greeneggies** [comment]
@valaparthvi can you try and reproduce this again?

**Comment by samjustus** [comment]
@valaparthvi can you try and reproduce this again? we think we've resolved the issue

**Comment by cpinjani** [comment]
The issue not seems fixed and is reproducible on latest head build:
![image](https://github.com/user-attachments/assets/87ecce16-6b7b-44a5-ac53-4e5b99f595c5)

Cluster provisioning status:
![image](https://github.com/user-attachments/assets/6b3df076-4d09-42be-bfcf-bce4960cd3ce)

Rancher tag:
![image](https://github.com/user-attachments/assets/128b1ea2-9a8f-4d62-8dc2-9a8bb5345cd1)

**Comment by mmartin24** [comment]
> Fleet v2.9-head e2e testing also affected when importing clusters into rancher.
> 
> Seems there is now a need to install Rancher `v2.9-head` with these additional helm flags:
> 
> ```
> --set extraEnv[0].name=CATTLE_AGENT_IMAGE \
> --set extraEnv[0].value=rancher/rancher-agent:v2.9-head
> ```
> 
> otherwise the image tag is not present in rancher import yaml for the `cattle-cluster-agent` Deployment resource and downstream cluster import will fail.
> 
> ```yaml
> ...
>           - name: CATTLE_INGRESS_IP_DOMAIN
>             value: sslip.io
>           - name: STRICT_VERIFY
>             value: "true"
>           image: rancher/rancher-agent:    <---- HERE
>           volumeMounts:
>           - name: cattle-credentials
>             mountPath: /cattle-credentials
>             readOnly: true
> ...
> ```
> 
> Not sure this is expected but few days back the extra flags were not needed.

FTR: adding here how how it looks when you try to provision downstream clusters without these flags (which point to regression):

![image (5)](https://github.com/user-attachments/assets/088410e4-28cf-4227-a228-fbee29516158)

**Comment by samjustus** [comment]
@cpinjani @valaparthvi can you try again?

**Comment by cpinjani** [comment]
To re-validate the issue, still don't see latest `2.9-head` image being created.

![image](https://github.com/user-attachments/assets/5ec42f9a-3343-4f07-9b41-5d64c26dfd6f)

**Comment by gaktive** [comment]
UI QA waiting on this to be resolved.

**Comment by snasovich** [comment]
Same as https://github.com/rancher/rancher/issues/46160#issuecomment-2231893244, ready to retest.

**Comment by cpinjani** [comment]
Validation passed, issue is fixed on build `v2.9-81337b95660cba868629d6fe769bd7e8242b5aee-head`
Able to provision/import hosted downstream clusters.

![image](https://github.com/user-attachments/assets/b2c8af43-4b07-4acb-8125-31035a6f0804)

![image](https://github.com/user-attachments/assets/859cdb50-7ebf-4ba6-8ee5-e653a944709c)

**Comment by mmartin24** [comment]
FYI: fleet able to provision downstream clusters as well on `v2.9-81337b95660cba868629d6fe769bd7e8242b5aee-head`:
![image](https://github.com/user-attachments/assets/7e51fe36-8f70-4ed5-8d48-e324001a8f32)
